### DISCUSSION TITLE
Enquiry about the Seedstudio sensecap indicator

### DISCUSSION THREAD

**Discussion by kraeml** [arendst/Tasmota#21815] [Q&A]
I have purchased the following display:

* https://www.seeedstudio.com/SenseCAP-Indicator-D1L-p-5646.html

I would like to operate it with HASPmota. So here's the question, does anyone have any experience with it? The display runs under OpenHasp. 

The ini file of OpenHasp can be found here:

* https://raw.githubusercontent.com/HASwitchPlate/openHASP/master/user_setups/esp32s3/sensecap-indicator.ini

I only got to know HASPmota in the last few days. So I'm new here.

Thanks for your help

**Comment by s-hadinger** [comment]
First thing to know is if it's octal or quad PSRAM. Looking at the template it's octal: `qio_opi`

I would start with the following `display.ini` from `S3-4848S040` as they look similar (I updated GPIOs, but not init sync)

```
:H,ST7701,480,480,16,RGB,18,17,16,21,45,15,14,13,12,11,10,9,8,7,6,5,4,3,2,1,0,6
:V,1,10,8,50,1,10,8,20,0
:S,2,1,1,0,40,20
:IS,41,48,-1,-1
FF,5,77,01,00,00,10
C0,2,3B,00
C1,2,0D,02
C2,2,31,05
CD,1,00
B0,10,00,11,18,0E,11,06,07,08,07,22,04,12,0F,AA,31,18
B1,10,00,11,19,0E,12,07,08,08,08,22,04,11,11,A9,32,18
FF,5,77,01,00,00,11
B0,1,60
B1,1,32
B2,1,07
B3,1,80
B5,1,49
B7,1,85
B8,1,21
C1,1,78
C2,1,78
E0,3,00,1B,02
E1,B,08,A0,00,00,07,A0,00,00,00,44,44
E2,C,11,11,44,44,ED,A0,00,00,EC,A0,00,00
E3,4,00,00,11,11
E4,2,44,44
E5,10,0A,E9,D8,A0,0C,EB,D8,A0,0E,ED,D8,A0,10,EF,D8,A0
E6,4,00,00,11,11
E7,2,44,44
E8,10,09,E8,D8,A0,0B,EA,D8,A0,0D,EC,D8,A0,0F,EE,D8,A0
EB,7,02,00,E4,E4,88,00,40
EC,2,3C,00
ED,10,AB,89,76,54,02,FF,FF,FF,FF,FF,FF,20,45,67,98,BA
FF,5,77,01,00,00,13
E5,1,E4
FF,5,77,01,00,00,00
21,0
3A,1,60
11,80
29,80
20,0
:B,120,02
```

Like all displays, you need to have a GPIO with `OPTIONA-3` configured

**Reply by sfromis** [reply]
Looking at the block diagram in the user manual, I see 4 pins for the display part coming from a PCA9535W expander. Two of them (CS/REST) have `-1` in the openHASP config you posted, indicating that they are not essential, and two seems to be for touch INT/REST which again have `-1`. Of course, sensors or SD card connected to the RP2040 will not be directly visible to Tasmota on the ESP32-S3.

**Reply by kraeml** [reply]
Oh, great tip. Thank you. I realise that the RP2040 cannot be addressed by Tasmota. I'm primarily interested in the display, touch and yes, at some point also Lorawan and SD card.

**Reply by sfromis** [reply]
It should be doable to talk to the RP2040 via serial, if you know the protocol, and what you can do there.

**Comment by kraeml** [comment]
Thanks for the quick reply. The sensecap should have 8MB flash.
I have also arrived at the S3-4848s040. I'm just starting to understand the display.ini.
The documentation is here: https://tasmota.github.io/docs/Universal-Display-Driver/#descriptor-file
I now understand the header (:H) better.
The video signal (:V) is also clearer now.
The splash (:S) seems to be standard.

The SPI initialisation (:IS) is still a bit unclear to me. Let's see how Openhasp does it. I assume that the following lines are the INIT setup.

The :B LVGL section is standard, isn't it?

Then I can start with the GPIO template. Thanks, that helped me a lot.

Off topic: Why does the green in RGB have one bit more than the others?

Translated with DeepL.com (free version)

**Reply by sfromis** [reply]
The extra bit for green is due to 16 bits meaning that one of RGB can get the extra bit, and the human eye has greater sensitivity to gradations of green.

**Comment by kraeml** [comment]
Ok, unfortunately my first template did not work. Here is the output on the tasmota console:
```
00:00:00.253 CMD: Fall back to serial port, no SOF packet detected on USB port
00:00:00.254 HDW: ESP32-S3 v0.2 
00:00:00.294 UFS: FlashFS mounted with 4396 kB free
00:00:00.311 CFG: Loaded from File, Count 24
00:00:00.312 FRC: Some settings have been reset (4)
00:00:00.318 SPI: Bus1 using GPIO41(CLK), GPIO48(MOSI) and GPIO47(MISO)
00:00:00.320 I2C: Bus1 using GPIO40(SCL) and GPIO39(SDA)
00:00:00.339 Project tasmota - Tasmota Version 14.1.0.4(f42e44a-lvgl-haspmota)-3_0_2(2024-07-22T13:59:08)
00:00:00.370 I2C: ADS1115 found at 0x48
00:00:01.000 WIF: Connecting to AP1 Sesamstreet Channel 1 BSSId 34:31:C4:CB:A8:E8 in mode HT20 as tasmota-33EBC0-3008...
00:00:02.727 WIF: Connected
00:00:03.014 HTP: Web server active on tasmota-33EBC0-3008 with IP address 192.168.56.194
21:00:35.088 RSL: INFO1 = {"Info1":{"Module":"ESP32S3-sensecap","Version":"14.1.0.4(f42e44a-lvgl-haspmota)","FallbackTopic":"cmnd/DVES_33EBC0_fb/","GroupTopic":"cmnd/tasmotas/"}}
21:00:35.101 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-33EBC0-3008","IPAddress":"192.168.56.194","IP6Global":"","IP6Local":"fe80::3230:f9ff:fe33:ebc0%st1"}}
21:00:35.114 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":28,"Reason":"LoadProhibited","EPC":"40057143","EXCVADDR":"00000000","CallChain":["40057140","42054802","420551ff","420645e4","4202dc43","4206455f","4200ebd4","42008bae","420cc21e"]},"BootCount":9}}
21:00:35.136 RSL: RESULT = {"POWER":"OFF"}
21:00:35.137 RSL: POWER = OFF
21:00:37.689 QPC: Reset
21:00:39.687 RSL: STATE = {"Time":"2024-07-22T21:00:39","Uptime":"0T00:00:09","UptimeSec":9,"Heap":217,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":65,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Sesamstreet","BSSId":"34:31:C4:CB:A8:E8","Channel":1,"Mode":"HT20","RSSI":100,"Signal":-30,"LinkCount":1,"Downtime":"0T00:00:04"}}
21:00:39.786 RSL: SENSOR = {"Time":"2024-07-22T21:00:39","ADS1115":{"A0":0,"A1":0,"A2":0,"A3":0}}
21:00:49.278 CMD: template
21:00:49.283 RSL: RESULT = {"NAME":"ESP32S3-sensecap","GPIO":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,6210,0,0,0,0,32,640,608,736,0,0,0,0,0,672,704],"FLAG":0,"BASE":1}
21:02:07.166 RSL: RESULT = {"POWER":"ON"}
21:02:07.167 RSL: POWER = ON
```
I've already found something, the button and hey, an ADC on the I2C. I'm quite proud of the first one, but unfortunately the display is blank and dark. If someone writes now, of course because ... then I would be happy. Otherwise I'll have to go back to it.  And at second glance, the PSRAM is missing.

But thanks again, it has helped me a lot so far.

Kraeml

**Reply by s-hadinger** [reply]
I have ordered one, stay tuned

**Reply by sfromis** [reply]
You have a more basic problem to be fixed before focusing on making the display work.
> `FRC: Some settings have been reset (4)`

This indicates that Tasmota is crashing quickly after boot, and therefore starts to stepwise restore defaults until able to boot without crashing again. This is related to that:
> `"RestartReason":{"Exception":28,"Reason":"LoadProhibited","EPC"` ...

Maybe more clues about the crashing if you follow the log via serial while booting.

And not getting PSRAM suggests that you're not using a binary including this feature. If you're not already using this one, try that.
> https://github.com/tasmota/install/raw/firmware/firmware/unofficial/tasmota32s3-opi_opi.factory.bin

Or the one without `.factory` for OTA upgrade, instead of reflashing from scratch.

**Reply by kraeml** [reply]
Oh, that's not nice to hear. I have reinstalled the LVGL. Here is the output:

```
00:00:00.252 CMD: Fall back to serial port, no SOF packet detected on USB port
00:00:00.253 HDW: ESP32-S3 v0.2 
00:00:00.264 UFS: FlashFS mounted with 4404 kB free
00:00:00.276 CFG: Loaded from File, Count 5
00:00:00.285 QPC: Count 1
00:00:00.329 BRY: Berry initialized, RAM used 5309 bytes
00:00:00.347 Project tasmota - Tasmota Version 14.1.0.4(895fbb2-lvgl-haspmota)-3_0_2(2024-07-23T14:02:04)
00:00:01.001 WIF: Connecting to AP1 Sesamstreet in mode HT20 as tasmota-33EBC0-3008...
00:00:02.614 WIF: Connected
20:39:53.115 HTP: Web server active on tasmota-33EBC0-3008 with IP address 192.168.56.194
20:39:54.164 RSL: INFO1 = {"Info1":{"Module":"ESP32S3","Version":"14.1.0.4(895fbb2-lvgl-haspmota)","FallbackTopic":"cmnd/DVES_33EBC0_fb/","GroupTopic":"cmnd/tasmotas/"}}
20:39:54.178 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-33EBC0-3008","IPAddress":"192.168.56.194","IP6Global":"2003:ea:f17:6c00:3230:f9ff:fe33:ebc0","IP6Local":"fe80::3230:f9ff:fe33:ebc0%st1"}}
20:39:54.193 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":7,"Reason":"PCValue","EPC":"4211cce6","EXCVADDR":"00000000","CallChain":[]},"BootCount":2}}
20:39:56.684 QPC: Reset
20:39:58.693 RSL: STATE = ...
20:42:16.830 CMD: template
20:42:16.834 RSL: RESULT = {"NAME":"ESP32S3","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":1}
```

No more error message with FRC.

My question now would be, how do I get LVGL with PSRAM?

@s-hadinger THX I stay tuned.

Kraeml

**Reply by sfromis** [reply]
You are still having some trouble with a crash (IDK what happens):
`{"Info3":{"RestartReason":{"Exception":7,"Reason":"PCValue","EPC"` ...

If you tried the binary I linked, try changing the name to replace `opi_opi` with `qio_opi`.

**Reply by kraeml** [reply]
Ok web installer with qio_opi works:

```
00:00:00.251 CMD: Fall back to serial port, no SOF packet detected on USB port
00:00:00.251 HDW: ESP32-S3 v0.2 (PSRAM)
00:00:00.259 UFS: FlashFS mounted with 4400 kB free
00:00:00.265 CFG: Loaded from File, Count 7
00:00:00.270 QPC: Count 1
00:00:00.285 BRY: Berry initialized, RAM used 5309 bytes
00:00:00.290 Project tasmota - Tasmota Version 14.1.0.4(895fbb2-lvgl-haspmota)-3_0_2(2024-07-23T14:03:20)
00:00:01.001 WIF: Connecting to AP1 Sesamstreet in mode HT20 as tasmota-33EBC0-3008...
00:00:02.738 WIF: Connected
00:00:02.940 HTP: Web server active on tasmota-33EBC0-3008 with IP address 192.168.56.194
23:04:23.204 RSL: INFO1 = {"Info1":{"Module":"ESP32S3","Version":"14.1.0.4(895fbb2-lvgl-haspmota)","FallbackTopic":"cmnd/DVES_33EBC0_fb/","GroupTopic":"cmnd/tasmotas/"}}
23:04:23.215 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-33EBC0-3008","IPAddress":"192.168.56.194","IP6Global":"2003:ea:f17:6c00:3230:f9ff:fe33:ebc0","IP6Local":"fe80::3230:f9ff:fe33:ebc0%st1"}}
23:04:23.236 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":4}}
23:04:25.532 QPC: Reset
23:04:27.530 RSL: STATE = ...
23:04:58.953 CMD: template
23:04:58.955 RSL: RESULT = {"NAME":"ESP32S3","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":1}
```
Thx

**Reply by kraeml** [reply]
#Forget this I found Main/Configure Other and activate the template. So I2Cscan is runnig.

So, I want to go through this step by step and have set the I2C to GPIO39 SDA and GPIO40 to SCL. Unfortunately, the I2Cscan does not work. Also, the settings are not written permanently. This means that after seven reboots, Wlan must be set and the template reset. I have not yet found the cause. Here is the console:
```
00:00:00.251 CMD: Fall back to serial port, no SOF packet detected on USB port
00:00:00.252 HDW: ESP32-S3 v0.2 (PSRAM)
00:00:00.258 UFS: FlashFS mounted with 4400 kB free
00:00:00.264 CFG: Loaded from File, Count 10
00:00:00.268 QPC: Count 1
00:00:00.282 BRY: Berry initialized, RAM used 5309 bytes
00:00:00.287 Project tasmota - Tasmota Version 14.1.0.4(895fbb2-lvgl-haspmota)-3_0_2(2024-07-23T14:03:20)
00:00:01.000 WIF: Connecting to AP1 Sesamstreet Channel 1 BSSId 34:31:C4:CB:A8:E8 in mode HT20 as tasmota-33EBC0-3008...
00:00:02.687 WIF: Connected
00:00:02.940 HTP: Web server active on tasmota-33EBC0-3008 with IP address 192.168.56.194
08:30:56.424 RSL: INFO1 = {"Info1":{"Module":"ESP32S3","Version":"14.1.0.4(895fbb2-lvgl-haspmota)","FallbackTopic":"cmnd/DVES_33EBC0_fb/","GroupTopic":"cmnd/tasmotas/"}}
08:30:56.435 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-33EBC0-3008","IPAddress":"192.168.56.194","IP6Global":"2003:ea:f10:eb00:3230:f9ff:fe33:ebc0","IP6Local":"fe80::3230:f9ff:fe33:ebc0%st1"}}
08:30:56.456 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":5}}
08:30:58.491 QPC: Reset
...
08:31:17.845 CMD: template
08:31:17.847 RSL: RESULT = {"NAME":"ESP32S3-sensecap","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,1,640,608,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":1}
08:31:24.797 CMD: i2cscan0
08:31:24.798 RSL: RESULT = {"Command":"Error"}
08:31:30.847 CMD: i2cscan
08:31:30.848 RSL: RESULT = {"Command":"Error"}
```

I have also set up the display with esphome. The i2c scan works.  However, the display does not.

Kraeml

**Reply by sfromis** [reply]
Until you have the I2C pins configured, I2CScan will fail. And the above shows that you do **not** have those pins configured, as the template is not activated. One of way of activating it is the command `Module 0`. You can also select the entry in "Configure Module", or check the box on "Configure Other".

**Reply by kraeml** [reply]
Thanks, I found it too. Sorry, I'm new to Tasmota, but thanks to the help here I understand better and better. So I2C, SPI(?), backlight and buttons work for me. I'll have another look at the display.ini. Or does anyone else see an error here?

```
00:00:00.251 CMD: Fall back to serial port, no SOF packet detected on USB port
00:00:00.252 HDW: ESP32-S3 v0.2 (PSRAM)
00:00:00.258 UFS: FlashFS mounted with 4400 kB free
00:00:00.264 CFG: Loaded from File, Count 43
00:00:00.269 QPC: Count 1
00:00:00.271 I2C: Bus1 using GPIO40(SCL) and GPIO39(SDA)
00:00:00.287 BRY: Berry initialized, RAM used 5309 bytes
00:00:00.295 Project tasmota - Sensecap Version 14.1.0.4(895fbb2-lvgl-haspmota)-3_0_2(2024-07-23T14:03:20)
00:00:00.321 I2C: ADS1115 found at 0x48
00:00:01.000 WIF: Connecting to AP1 Sesamstreet Channel 1 BSSId 34:31:C4:CB:A8:E8 in mode HT20 as tasmota-33EBC0-3008...
00:00:02.843 WIF: Connected
00:00:03.071 HTP: Web server active on tasmota-33EBC0-3008 with IP address 192.168.56.194
09:09:48.070 RSL: INFO1 = {"Info1":{"Module":"ESP32S3-sensecap","Version":"14.1.0.4(895fbb2-lvgl-haspmota)","FallbackTopic":"cmnd/DVES_33EBC0_fb/","GroupTopic":"cmnd/tasmotas/"}}
09:09:48.082 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-33EBC0-3008","IPAddress":"192.168.56.194","IP6Global":"2003:ea:f10:eb00:3230:f9ff:fe33:ebc0","IP6Local":"fe80::3230:f9ff:fe33:ebc0%st1"}}
09:09:48.103 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":6}}
09:09:48.104 RSL: RESULT = {"POWER":"ON"}
09:09:48.115 RSL: POWER = ON
09:09:50.539 QPC: Reset
...
09:09:52.641 RSL: SENSOR = {"Time":"2024-07-24T09:09:52","ADS1115":{"A0":0,"A1":0,"A2":0,"A3":0}}
09:10:13.091 CMD: template
09:10:13.094 RSL: RESULT = {"NAME":"ESP32S3-sensecap","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,6210,0,0,0,0,32,640,608,896,1,1,1,416,1,672,704],"FLAG":0,"BASE":1}
09:11:53.178 RSL: RESULT = {"POWER":"OFF"}
09:11:53.178 RSL: POWER = OFF
09:11:54.939 RSL: RESULT = {"POWER":"ON"}
09:11:54.939 RSL: POWER = ON
```

Kraeml

**Reply by kraeml** [reply]
Ok, with the display.ini from above I get the following error message:
```
00:00:00.251 CMD: Fall back to serial port, no SOF packet detected on USB port
00:00:00.251 HDW: ESP32-S3 v0.2 (PSRAM)
00:00:00.256 UFS: FlashFS mounted with 4396 kB free
00:00:00.261 CFG: Loaded from File, Count 59
00:00:00.262 FRC: Some settings have been reset (4)
00:00:00.274 I2C: Bus1 using GPIO40(SCL) and GPIO39(SDA)
00:00:00.279 Project tasmota - Sensecap Version 14.1.0.4(895fbb2-lvgl-haspmota)-3_0_2(2024-07-23T14:03:20)
00:00:00.305 I2C: ADS1115 found at 0x48
00:00:01.001 WIF: Connecting to AP1 Sesamstreet Channel 1 BSSId 34:31:C4:CB:A8:E8 in mode HT20 as tasmota-33EBC0-3008...
00:00:02.860 WIF: Connected
00:00:03.135 HTP: Web server active on tasmota-33EBC0-3008 with IP address 192.168.56.194
09:41:11.379 RSL: INFO1 = {"Info1":{"Module":"ESP32S3-sensecap","Version":"14.1.0.4(895fbb2-lvgl-haspmota)","FallbackTopic":"cmnd/DVES_33EBC0_fb/","GroupTopic":"cmnd/tasmotas/"}}
09:41:11.390 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-33EBC0-3008","IPAddress":"192.168.56.194","IP6Global":"2003:ea:f10:eb00:3230:f9ff:fe33:ebc0","IP6Local":"fe80::3230:f9ff:fe33:ebc0%st1"}}
09:41:11.412 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":28,"Reason":"LoadProhibited","EPC":"40057143","EXCVADDR":"00000000","CallChain":["40057140","420544ae","42054eab","420697e0","4202dd8b","42069f4f","4200ed40","42008c2e","420d18be"]},"BootCount":7}}
09:41:11.433 RSL: RESULT = {"POWER":"ON"}
09:41:11.433 RSL: POWER = ON
09:41:14.505 QPC: Reset
...
09:41:15.637 RSL: SENSOR = {"Time":"2024-07-24T09:41:15","ADS1115":{"A0":0,"A1":0,"A2":0,"A3":0}}
09:44:21.574 CMD: template
09:44:21.576 RSL: RESULT = {"NAME":"ESP32S3-sensecap","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,6210,0,0,0,0,32,640,608,896,1,1,1,416,1,672,704],"FLAG":0,"BASE":1}
```

Thanks Kraeml

**Reply by sfromis** [reply]
Now you're back to the bad situation with:
`FRC: Some settings have been reset (4)`
and
`{"Info3":{"RestartReason":{"Exception":28,"Reason":"LoadProhibited",` ...

I can't say what happens, but as already mentioned, following the failing boots on serial may get you closer to a clue. Speculatively, maybe a bad display setup causes a failure.

**Comment by Jason2866** [comment]
The block circuit diagram. Beside of this no useful info from manufacturer. Many pages of marketing...
<img width="1008" alt="image" src="https://github.com/user-attachments/assets/27e772f1-e5f6-4ceb-815e-552ba6212a70">

**Comment by Jason2866** [comment]
The inbuilt sensors and the SD Card will **not** be usable with Tasmota

@s-hadinger Do I have to think about RP2040 Arduino Platformio support setup? If you write a onboard Tasmota RP2040 Flasher ;-)

**Reply by s-hadinger** [reply]
Not there yet. Actually I'm not sure what the RP2040 would be useful to a Tasmota setup

**Reply by kraeml** [reply]
Yes, that's clear. But that's not the task right now. I think something will be found. Again, yes, the block diagram is not good and the documentation is very poor in the PDFs.

**Reply by Jason2866** [reply]
There is a ";-)" in my post. So kidding, not really ment serious. But I have a RP2040 clone lying around :-)

**Reply by sfromis** [reply]
In that specific setup, you *could* talk to the RP2040 via the indicated serial connection, and then getting sensor/SD/buzzer access would "not be impossible", for someone really wanting to "have fun". However, it does not seem "very practical".... Still, it does appear that connecting stuff (like sensors) directly to the ESP32-S3 is not provided for.

For Tasmota, the device makes sense if what you want is ESP32-S3 with a display and LoRa, but other than that - not so much. Note that you may not get LoRa to work with Tasmota, as some of the pins for that is via the IO Expander chip.

**Reply by Jason2866** [reply]
> For Tasmota, the device makes sense if what you want is ESP32-S3 with a display and LoRa, but other than that - not so much.

Totally agree with this. Honestly I do find the device a bit expensive for the hardware in there.

**Reply by kraeml** [reply]
FYI: Here a project for sensecap https://github.com/YooSHome/SenseCapConfig The SD Card configure the lvgl/esp ;-)

**Reply by Jason2866** [reply]
> FYI: Here a project for sensecap https://github.com/YooSHome/SenseCapConfig The SD Card configure the lvgl/esp ;-)

Tasmota is not Home Assistant. To be clear. The SD Card will not be usable with Tasmota!!

**Reply by kraeml** [reply]
Sure, otherwise Tasmota would be called Homemota ;-) No seriously, the RP2040, SD card etc. is not my concern for the time being and in any case has nothing to do with Tasmota. The Sensecap with Haspmota would be quite good. I think with Berry and some ‘fun’ (I'm currently trying out the docker compile from Tasmota) you can also get Lora to work. The link was just for general information about what others are doing with the box.

**Comment by s-hadinger** [comment]
I finally got back to it. I got the parameters from OpenHASP. The display now works, I still need to work on the touch screen.

`display.ini`: 
```
:H,ST7701,480,480,16,RGB,18,17,16,21,45,15,14,13,12,11,10,9,8,7,6,5,4,3,2,1,0,6
:V,1,10,8,50,1,10,8,20,0
:S,2,1,1,0,40,20
:IS,41,48,-1,-1
FF,5,77,01,00,00,10
C0,2,3B,00
C1,2,0D,02
C2,2,31,05
C7,1,04
CD,1,08
B0,10,00,11,18,0E,11,06,07,08,07,22,04,12,0F,AA,31,18
B1,10,00,11,19,0E,12,07,08,08,08,22,04,11,11,A9,32,18
FF,5,77,01,00,00,11
B0,1,60
B1,1,32
B2,1,07
B3,1,80
B5,1,49
B7,1,85
B8,1,21
C1,1,78
C2,A1,78
E0,3,00,1B,02
E1,B,08,A0,00,00,07,A0,00,00,00,44,44
E2,C,11,11,44,44,ED,A0,00,00,EC,A0,00,00
E3,4,00,00,11,11
E4,2,44,44
E5,10,0A,E9,D8,A0,0C,EB,D8,A0,0E,ED,D8,A0,10,EF,D8,A0
E6,4,00,00,11,11
E7,2,44,44
E8,10,09,E8,D8,A0,0B,EA,D8,A0,0D,EC,D8,A0,0F,EE,D8,A0
EB,7,02,00,E4,E4,88,00,40
EC,2,3C,00
ED,10,AB,89,76,54,02,FF,FF,FF,FF,FF,FF,20,45,67,98,BA
36,1,10
FF,5,77,01,00,00,13
E1,1,E4
FF,5,77,01,00,00,00
21,0
3A,1,50
11,80
29,80
:B,120,02
#
```

Edit, here is the template:
```
Template {"NAME":"SenseCAP Indicator D1","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,3200,3232,0,6210,0,0,0,0,32,640,608,0,1,1,1,0,1,0,0],"FLAG":0,"BASE":1}
```

**Reply by kraeml** [reply]
Yes, no errors. Thanks

**Comment by s-hadinger** [comment]
The full configuration is described here: https://tasmota.github.io/docs/devices/SeedStudio-SenseCAP-D1/

With full Air Quality sensor support

![Web Console showing readings](https://tasmota.github.io/docs/_media/devices/SeedStudio-SenseCap-D1-Console.png)
